packages/network-controller/src/NetworkController.ts

```diff
@@ -391,9 +389,9 @@ export class NetworkController extends BaseControllerV2<
     this.#registerProvider();
   }
 
-  #safelyStopProvider(provider: Provider | undefined) {
+  #safelyStopProvider() {
     setTimeout(() => {
-      provider?.stop();
+      this.#providerProxy?.stop();
```

Comment:
Storing a reference to the provider is useful because by the time this function is run, the provider may have changed. Once we don't need to stop the provider, then we really don't need `this.#provider`.